plugin/ochttp/trace_test.go

```diff
@@ -173,116 +173,170 @@ func (c *collector) ExportSpan(s *trace.SpanData) {
 }
 
 func TestEndToEnd(t *testing.T) {
-	var spans collector
-	trace.RegisterExporter(&spans)
-	defer trace.UnregisterExporter(&spans)
-
-	span := trace.NewSpan(
-		"top-level",
-		nil,
-		trace.StartOptions{
-			Sampler: trace.AlwaysSample(),
-		})
-	ctx := trace.WithSpan(context.Background(), span)
-
-	serverDone := make(chan struct{})
-	serverReturn := make(chan time.Time)
-	url := serveHTTP(serverDone, serverReturn)
-
-	req, err := http.NewRequest(
-		"POST",
-		fmt.Sprintf("%s/example/url/path?qparam=val", url),
-		strings.NewReader("expected-request-body"))
-	if err != nil {
-		t.Fatalf("unexpected error %#v", err)
-	}
-	req = req.WithContext(ctx)
-
-	rt := &Transport{
-		NoStats:     true,
-		Propagation: defaultFormat,
-		Base:        http.DefaultTransport,
-	}
-	resp, err := rt.RoundTrip(req)
-	if err != nil {
-		t.Fatalf("unexpected error %s", err)
-	}
-	if resp.StatusCode != http.StatusOK {
-		t.Fatalf("unexpected stats: %d", resp.StatusCode)
+	trace.SetDefaultSampler(trace.AlwaysSample())
+
+	cases := []struct {
+		Name              string
+		ServerMiddleware  *Handler
+		ClientMiddleware  *Transport
+		ExpectSameTraceID bool
+		ExpectLinks       bool // expect a link between client and server span
+	}{
+		{
+			Name:              "internal default propagation",
+			ServerMiddleware:  &Handler{},
+			ClientMiddleware:  &Transport{NoStats: true},
+			ExpectSameTraceID: true,
+		},
+		{
+			Name:              "external default propagation",
+			ServerMiddleware:  &Handler{IsPublicEndpoint: true},
+			ClientMiddleware:  &Transport{NoStats: true},
+			ExpectSameTraceID: false,
+			ExpectLinks:       true,
+		},
+		{
+			Name:              "internal TraceContext propagation",
+			ServerMiddleware:  &Handler{Propagation: &tracecontext.HTTPFormat{}},
+			ClientMiddleware:  &Transport{NoStats: true, Propagation: &tracecontext.HTTPFormat{}},
+			ExpectSameTraceID: true,
+		},
+		{
+			Name:              "misconfigured propagation",
+			ServerMiddleware:  &Handler{IsPublicEndpoint: true, Propagation: &tracecontext.HTTPFormat{}},
+			ClientMiddleware:  &Transport{NoStats: true, Propagation: &b3.HTTPFormat{}},
+			ExpectSameTraceID: false,
+			ExpectLinks:       false,
+		},
 	}
 
-	serverReturn <- time.Now().Add(time.Millisecond)
+	for _, c := range cases {
```

Comment:
In Go, it is canonical to name cases tc, and specific case tt.

It improves readability to use tt because everyone knows it represents the current test case without having to jump to the declaration.